UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/Helper/AuthTokenProvider.cs

```diff
@@ -0,0 +1,126 @@
+﻿using JWT;
+using JWT.Algorithms;
+using JWT.Serializers;
+using Microsoft.Identity.Client;
+using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+using System.Threading.Tasks;
+using static UKHO.ExchangeSetService.API.FunctionalTests.Helper.TestConfiguration;
+
+namespace UKHO.ExchangeSetService.API.FunctionalTests.Helper
+{
+    public class AuthTokenProvider
+    {
+        static string EssAccessToken = null;
+        static string EssAccessTokenNoAuth = null;
+        static EssAuthorizationTokenConfiguration EssauthConfig = new TestConfiguration().EssAuthorizationConfig;
+
+        public async Task<string> GetEssToken()
+        {
+            EssAccessToken = await GenerateEssToken(EssauthConfig.AutoTestClientId, EssauthConfig.AutoTestClientSecret, EssAccessToken);
+            return EssAccessToken;
+        }
+
+        public async Task<string> GetEssTokenNoAuth()
+        {
+            EssAccessTokenNoAuth = await GenerateEssToken(EssauthConfig.AutoTestClientIdNoAuth, EssauthConfig.AutoTestClientSecretNoAuth, EssAccessTokenNoAuth);
+            return EssAccessTokenNoAuth;
+        }
+
+        /// <summary>
+        /// GenerateToken
+        /// </summary>
+        /// <param name="ClientId"></param>
+        /// <param name="ClientSecrect"></param>
+        /// <param name="Token"></param>
+        /// <returns></returns>
+
+        private static async Task<string> GenerateEssToken(string ClientId, string ClientSecrect, string Token)
```

Comment:
Small typo: ClientSecrect should be ClientSecret

Reply:
I have done the changes. Thanks Martyn.